File: year2/systems-design-implementation/Lab3/Lab3Sharp/Repo/Database/ProbaRepositoryDB.cs
using Lab2Sharp.Domain;
using Lab2Sharp.Repo.Interfaces;
using log4net;
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2Sharp.Repo.Database
{
    class ProbaRepositoryDB<ID, E> : ProbaRepository<int, Proba>
    {
        private string url;
        private static readonly ILog log = LogManager.GetLogger("ParticipantRepository");

        public ProbaRepositoryDB(string url)
        {
            this.url = url;
        }

        public Proba FindOne(int id)
        {
            string sqlCom = "select * from Probe where id=@id";
            using (var conn = new SQLiteConnection(url))
            {
                conn.Open();
                using (var cmd = new SQLiteCommand(sqlCom, conn))
                {
                    var idd = cmd.CreateParameter();
                    idd.ParameterName = "@id";
                    idd.Value = id;
                    cmd.Parameters.Add(idd);
                    using (var data = cmd.ExecuteReader())
                    {
                        if (data.Read())
                        {
                            String stil = data.GetString(2);
                            int distanta = data.GetInt32(1);
                            Proba p = new Proba(id, distanta, stil);
                            return p;
                        }
                    }
                }
            }
            return null;
        }
        public IEnumerable<Proba> FindAll()
        {
            log.InfoFormat("Entering findAll");
            IList<Proba> list = new List<Proba>();
            string sqlCom = "select * from Probe";
            using (var conn = new SQLiteConnection(url))
            {
                conn.Open();
                var cmd = new SQLiteCommand(sqlCom, conn);
                using (var data = cmd.ExecuteReader())
                {
                    while (data.Read())
                    {
                        int id = data.GetInt32(0);
                        String stil = data.GetString(2);
                        int distanta = data.GetInt32(1);
                        Proba p = new Proba(id, distanta, stil);
                        log.InfoFormat("Found {0}", p);
                        list.Add(p);
                    }
                }
            }
            log.InfoFormat("Exiting findAll");
            return list;
        }
        public Proba Add(Proba entity)
        {
            Proba p = entity;
            string sqlCom = "insert into Probe (distanta, stil) values (@distanta,@stil)";
            using (var conn = new SQLiteConnection(url))
            {
                conn.Open();
                using (var cmd = new SQLiteCommand(sqlCom, conn))
                {
                    var distanta = cmd.CreateParameter();
                    distanta.ParameterName = "@distanta";
                    distanta.Value = entity.distanta;
                    cmd.Parameters.Add(distanta);
                    var stil = cmd.CreateParameter();
                    stil.ParameterName = "@stil";
                    stil.Value = entity.stil;
                    cmd.Parameters.Add(stil);
                    if (cmd.ExecuteNonQuery() > 0)
                        p = null;
                }
            }
            return p;
        }
        public Proba Delete(int id)
        {
            Proba p = FindOne(id);
            string sqlCom = "delete from Probe where id=@id";
            using (var conn = new SQLiteConnection(url))
            {
                conn.Open();
                using (var cmd = new SQLiteCommand(sqlCom, conn))
                {
                    var idd = cmd.CreateParameter();
                    idd.ParameterName = "@id";
                    idd.Value = id;
                    cmd.Parameters.Add(idd);
                    cmd.ExecuteNonQuery();
                }
            }
            return p;
        }
        public Proba Update(Proba entity)
        {
            Proba p = FindOne(entity.id);
            string sqlCom = "update Probe set distanta=@distanta,stil=@stil where id=@id";
            using (var conn = new SQLiteConnection(url))
            {
                conn.Open();
                using (var cmd = new SQLiteCommand(sqlCom, conn))
                {
                    var id = cmd.CreateParameter();
                    id.ParameterName = "@id";
                    id.Value = entity.id;
                    cmd.Parameters.Add(id);
                    var distanta = cmd.CreateParameter();
                    distanta.ParameterName = "@distanta";
                    distanta.Value = entity.distanta;
                    cmd.Parameters.Add(distanta);
                    var stil = cmd.CreateParameter();
                    stil.ParameterName = "@stil";
                    stil.Value = entity.stil;
                    cmd.Parameters.Add(stil);
                    cmd.ExecuteNonQuery();
                }
            }
            return p;
        }
    }
}

File: year2/systems-design-implementation/Lab3/Lab3/src/main/java/repo/database/ProbaRepositoryDB.java
package repo.database;

import domain.Participant;
import domain.Proba;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import repo.interfaces.ProbaRepository;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class ProbaRepositoryDB implements ProbaRepository<Integer, Proba> {
    private final static Logger logger = LogManager.getLogger(ParticipantRepositoryDB.class);
    private String url;

    public ProbaRepositoryDB(String url) {
        this.url = url;
    }

    @Override
    public Optional<Proba> findOne(Integer id) throws SQLException {
        logger.traceEntry();
        Optional<Proba> p = Optional.empty();
        Connection connection = DriverManager.getConnection(url);
        try (PreparedStatement preparedStatement = connection.prepareStatement("select * from Probe where id=?")) {
            preparedStatement.setInt(1, id);
            ResultSet resultSet = preparedStatement.executeQuery();
            while (resultSet.next()) {
                String stil = resultSet.getString("stil");
                int distanta = resultSet.getInt("distanta");
                p = Optional.of(new Proba(id, distanta, stil));
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println(ex);
        }
        logger.traceExit(p.get());
        return p;
    }

    @Override
    public Iterable<Proba> findAll() throws SQLException {
        logger.traceEntry();
        Connection connection = DriverManager.getConnection(url);
        List<Proba> list = new ArrayList<>();
        try (PreparedStatement preparedStatement = connection.prepareStatement("select * from Probe")) {
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                while (resultSet.next()) {
                    int id = resultSet.getInt("id");
                    String stil = resultSet.getString("stil");
                    int distanta = resultSet.getInt("distanta");
                    Proba p = new Proba(id, distanta, stil);
                    list.add(p);
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println(ex);
        }
        logger.traceExit(list);
        return list;
    }

    @Override
    public Optional<Proba> add(Proba entity) throws SQLException {
        logger.traceEntry("Adding task {} ", entity);
        Optional<Proba> p = Optional.of(entity);
        Connection connection = DriverManager.getConnection(url);
        try (PreparedStatement preStmt = connection.prepareStatement("insert into Probe (distanta, stil) values (?,?)")) {
            preStmt.setInt(1, entity.getDistanta());
            preStmt.setString(2, entity.getStil());
            int result = preStmt.executeUpdate();
            logger.trace("Added {} instance", result);
            p = Optional.empty();
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return p;
    }

    @Override
    public Optional<Proba> delete(Integer id) throws SQLException {
        logger.traceEntry();
        Optional<Proba> p = findOne(id);
        Connection connection = DriverManager.getConnection(url);
        try (PreparedStatement preparedStatement = connection.prepareStatement("delete from Probe where id=?")) {
            preparedStatement.setInt(1, id);
            preparedStatement.executeUpdate();
            logger.trace("Removed {} instance", p);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println(ex);
        }
        logger.traceExit();
        return p;
    }

    @Override
    public Optional<Proba> update(Proba entity) throws SQLException {
        logger.traceEntry();
        Optional<Proba> p = findOne(entity.getId());
        Connection connection = DriverManager.getConnection(url);
        try (PreparedStatement preparedStatement = connection.prepareStatement("update Probe set distanta=?,stil=? where id=?")) {
            preparedStatement.setInt(1, entity.getDistanta());
            preparedStatement.setString(2, entity.getStil());
            preparedStatement.setInt(3, entity.getId());
            preparedStatement.executeUpdate();
            logger.trace("Updated {} instance", p);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println(ex);
        }
        logger.traceExit();
        return p;
    }
}
